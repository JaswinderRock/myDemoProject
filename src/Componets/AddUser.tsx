import React, { Component, FormEvent } from 'react';
import '../css/model.css';
import CloseIcon from '../images/close.svg';
import { connect } from "react-redux";
import { addNewUser } from '../Redux/Actions/action';


interface AddUserState {
    firstName: string;
    lastName: string;
    email: string;
    companyName: string;
    occupation: string;
}

interface AddUserProps {
    addNewUser: (users: object) => void;
    showModal: boolean
    closeModel: () => void,
    userList: Array<{
        id: number
    }>
}


class AddUser extends Component<AddUserProps, AddUserState> {
    constructor(props: AddUserProps) {
        super(props);
        this.state = {
            firstName: '',
            lastName: '',
            email: '',
            companyName: '',
            occupation: '',
        };
    }
    handleSubmit = (event: FormEvent<HTMLFormElement>): void => {
        event.preventDefault();
        const { firstName, lastName, email, companyName, occupation } = this.state;
        const { addNewUser, userList, closeModel } = this.props;
        let obj = { firstName, lastName, email, companyName, occupation, id: userList[userList.length - 1].id + 1 }
        addNewUser(obj)
        closeModel()
    };
    render() {
        const { firstName, lastName, email, companyName, occupation } = this.state;
        return (
            <>
                <div className={`modal add-model ${this.props.showModal ? "show" : ""}`}>
                    <div className="modal__container modal__container" style={{ width: 545, background: 'white' }}>
                        <div
                            className="header"
                            style={{
                                paddingLeft: '20px',
                                paddingRight: '20px',
                                height: '30px',
                                borderBottom: "1px solid rgb(0 0 0 / 10%)",
                            }}
                        >
                            <div style={{ display: 'flex', justifyContent: 'space-between' }} >
                                <div style={{ marginTop: '-22px' }}>
                                    <h2 className="text-lg font-semibold">Add New User</h2>
                                </div>
                                <div className="modal__close-icon" style={{ cursor: 'pointer' }} onClick={this.props.closeModel}>
                                    <img src={CloseIcon} alt="" height={"18px"} width={"18px"} />
                                </div>
                            </div>

                        </div>
                        <div className="modal__body" style={{ padding: "20px" }}>
                            <form onSubmit={this.handleSubmit}>
                                <div>
                                    <div style={{ marginTop: '24px' }}>
                                        <label style={{ display: 'block', fontSize: '16px', color: '#374957', marginBottom: '10px', fontWeight: '600', fontFamily: 'sans-serif' }}>
                                            FirstName
                                        </label>
                                        <input
                                            style={{ width: '90%', padding: '10px', border: '1px solid #dedede', borderRadius: '4px', fontSize: '16px' }}
                                            placeholder="Enter firstName"
                                            type="text"
                                            id="firstName"
                                            name="firstName"
                                            value={firstName}
                                            onChange={e => this.setState({ firstName: e.target.value })}
                                            required
                                        />
                                    </div>
                                    <div style={{ marginTop: '24px' }}>
                                        <label style={{ display: 'block', fontSize: '16px', color: '#374957', marginBottom: '10px', fontWeight: '600', fontFamily: 'sans-serif' }}>
                                            LastName
                                        </label>
                                        <input
                                            style={{ width: '90%', padding: '10px', border: '1px solid #dedede', borderRadius: '4px', fontSize: '16px' }}
                                            placeholder='Enter lastName'
                                            type="text"
                                            id="lastName"
                                            name="lastName"
                                            value={lastName}
                                            onChange={e => this.setState({ lastName: e.target.value })}
                                            required
                                        />
                                    </div>
                                    <div style={{ marginTop: '24px' }}>
                                        <label style={{ display: 'block', fontSize: '16px', color: '#374957', marginBottom: '10px', fontWeight: '600', fontFamily: 'sans-serif' }}>
                                            Email
                                        </label>
                                        <input
                                            style={{ width: '90%', padding: '10px', border: '1px solid #dedede', borderRadius: '4px', fontSize: '16px' }}
                                            placeholder='Enter your email'
                                            type="email"
                                            id="email"
                                            name="email"
                                            value={email}
                                            onChange={e => this.setState({ email: e.target.value })}
                                            required
                                        />
                                    </div>
                                    <div style={{ marginTop: '24px' }}>
                                        <label style={{ display: 'block', fontSize: '16px', color: '#374957', marginBottom: '10px', fontWeight: '600', fontFamily: 'sans-serif' }}>
                                            CompanyName
                                        </label>
                                        <input
                                            style={{ width: '90%', padding: '10px', border: '1px solid #dedede', borderRadius: '4px', fontSize: '16px' }}
                                            placeholder='Enter your company name'
                                            type="text"
                                            id="companyName"
                                            name="companyName"
                                            value={companyName}
                                            onChange={e => this.setState({ companyName: e.target.value })}
                                            required
                                        />
                                    </div>
                                    <div style={{ marginTop: '24px' }}>
                                        <label style={{ display: 'block', fontSize: '16px', color: '#374957', marginBottom: '10px', fontWeight: '600', fontFamily: 'sans-serif' }}>
                                            Occupation
                                        </label>
                                        <input
                                            style={{ width: '90%', padding: '10px', border: '1px solid #dedede', borderRadius: '4px', fontSize: '16px' }}
                                            placeholder='Enter your occupation'
                                            type="text"
                                            id="occupation"
                                            name="occupation"
                                            value={occupation}
                                            onChange={e => this.setState({ occupation: e.target.value })}
                                            required
                                        />
                                    </div>
                                    <div style={{ display: 'flex', justifyContent: 'flex-end', marginTop: '40px' }}>
                                        <button
                                            className="btn cursor-pointer"
                                            type="button"
                                            name="cancle-btn"
                                            style={{ cursor: 'pointer', borderRadius: '4px', marginRight: '20px', marginLeft: '20px', border: '1px solid black', background: 'white', color: 'black' }}
                                            onClick={this.props.closeModel}
                                        >
                                            Cancel
                                        </button>
                                        <button
                                            className="btn ml-4 cursor-pointer"
                                            type="submit"
                                            style={{ cursor: 'pointer', borderRadius: '4px', marginRight: '20px', marginLeft: '20px', }}
                                        >
                                            Add User
                                        </button>
                                    </div>
                                </div>
                            </form>
                        </div>
                    </div >
                </div >
            </>
        )
    }
}
function mapStateToProps(state: any) {
    return { userList: state.userList };
}

export default connect(mapStateToProps, { addNewUser })(AddUser);