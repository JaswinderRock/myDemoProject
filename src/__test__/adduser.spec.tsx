import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../Redux/Store/store';
import AddUser from '../Componets/AddUser';
import { findByTestAttr } from '../Utils/index';

const setUp = () => {
    const component = mount(<Provider store={store}>
        <AddUser
            showModal={true} closeModel={() => { }}
        />
    </Provider>);
    return component;
}


describe('AddUser Component', () => {
    let component: any;
    beforeEach(() => {
        component = setUp();
    });
    test('renders without crashing', () => {
        expect(component.exists()).toBe(true);
    });
    test('should add snapshot ', () => {
        expect(component).toMatchSnapshot()
    });
    test('input fields change', () => {
        const firstName = findByTestAttr(component, 'firstName')
        expect(firstName.exists()).toBe(true);
        firstName.simulate('change', { target: { value: 'John' } })
        expect(firstName.exists()).toBe(true);
    });
    test('updates state when input fields firstName change', () => {
        const wrapper = setUp();
        expect(wrapper.find('input[name="firstName"]').prop('value')).toEqual('');
        const newFirstName = 'Test';
        const input = wrapper.find('input[name="firstName"]');
        input.simulate('change', {
            target: { name: 'firstName', value: newFirstName },
        });
        expect(wrapper.find('input[name="firstName"]').prop('value')).toEqual(newFirstName);
    });
    test('updates state when input fields LastName change', () => {
        const wrapper = setUp();
        expect(wrapper.find('input[name="lastName"]').prop('value')).toEqual('');
        const newLastName = 'Case';
        const input = wrapper.find('input[name="lastName"]');
        input.simulate('change', {
            target: { name: 'lastName', value: newLastName },
        });
        expect(wrapper.find('input[name="lastName"]').prop('value')).toEqual(newLastName);
    });
    test("Can change email field", () => {
        const wrapper = setUp();
        expect(wrapper.find('input[name="email"]').prop('value')).toEqual('');
        const newEmail = "test@gmail.com";
        const input = wrapper.find("input").at(0);
        input.simulate("change", {
            target: { name: "email", value: newEmail },
        });
        expect(input.exists()).toBe(true);
    })

    // it('calls addNewUser and closeModel when form is submitted', () => {
    //     const addNewUserMock = jest.fn();
    //     const closeModelMock = jest.fn();
    //     const wrapper = setUp();
    //     wrapper.find('#firstName').simulate('change', { target: { value: 'John' } });
    //     wrapper.find('form').simulate('submit');
    //     expect(addNewUserMock).toHaveBeenCalled();
    //     expect(closeModelMock).toHaveBeenCalled();
    // });

    // it('prevents form submission if required fields are empty', () => {
    //     const addNewUserMock = jest.fn();
    //     const closeModelMock = jest.fn();

    //     const wrapper = mount(
    //         <AddUser
    //             addNewUser={addNewUserMock}
    //             showModal={true}
    //             closeModel={closeModelMock}
    //             userList={[]}
    //         />
    //     );

    //     // Attempt to submit the form without filling in any fields
    //     wrapper.find('form').simulate('submit');

    //     // Verify that addNewUser and closeModel were not called
    //     expect(addNewUserMock).not.toHaveBeenCalled();
    //     expect(closeModelMock).not.toHaveBeenCalled();
    // });
});
