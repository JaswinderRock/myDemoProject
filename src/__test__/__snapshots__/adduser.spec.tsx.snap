// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser Component should add snapshot  1`] = `ReactWrapper {}`;
